mirrord/layer/src/error.rs

```diff
@@ -41,7 +41,7 @@ mod ignore_codes {
 /// These errors are converted to [`libc`] error codes, and are also used to [`set_errno`].
 #[derive(Error, Debug)]
 pub(crate) enum HookError {
-    #[error("mirrord-layer: Failed while getting a response!")]
+    #[error(transparent)]
```

Comment:
All other variants have `mirrord-layer` prefix :X

Reply:
Re-added it, but I think we should look into our error messages, while trying this out I was getting some terrible ones, along the lines of:

`mirrord-layer: Failed to get Response due to: Failed to DNS something with: KubeError failed { kind: Thing, error: UnknownSomething }`.

We should probably prepare a good initial message for the inner error, and just keep everything else `transparent` (for our internal errors), example:

```rust
#[error("We have an io error: {0}!")]
IO(#[from] std::IO),

#[error(transparent)]
Response(#[from] protocol::ResponseError),
```

Reply:
Yeah these error messages require some extra care